demo/backend/include/Options.hpp

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include "Palette.hpp"
+
+class Options
+{
+    private:
+        // realize singleton pattern
+        static Options* instance;
+        Options();
+
+
+        // private data members
+        Palette activePalette{};
+        Color bgColor{0xFFFFFF};
+
+    public:
+        // singleton should neither be copied nor assigned
```

Comment:
then you should also delete the move assigment and move constructor
`Options(Options&&)` `Options&  operator=(Options&&)`

Reply:
true; forgot about those